lib/jsonapi/authorization/authorizing_processor.rb

```diff
@@ -249,6 +249,8 @@ def related_models
         [:to_one, :to_many].flat_map do |rel_type|
           data[rel_type].flat_map do |assoc_name, assoc_value|
             case assoc_value
+            when nil
+              next
```

Comment:
I don't think we need to change the `def related_models` method at all. Your bug fix should only be necessary to touch the `related_models_with_context` method.